### PR TITLE
add road properties PATCH route

### PR DESCRIPTION
**Pull request by jameslaneconkling**
api changes for https://github.com/orma/openroads-vn-analytics/issues/209

### CODE REVIEW

File: routes/road-properties.js [hunk 1]
```diff
@@ -270,6 +273,34 @@ function deleteHandler(req, res) {
   });
 }
 
+
+function patchPropertyHandler(req, res) {
+  if (validate(req.payload) !== undefined) {
+    return res(Boom.badData());
+  }
```
Comment (mileswwatkins) [inline routes/road-properties.js]: Another check you may want to add is that the JSON object in the request is of depth 1, and doesn't have any nested objects/arrays.

File: routes/road-properties.js [hunk 2]
```diff
@@ -270,6 +273,34 @@ function deleteHandler(req, res) {
...
+    .select('id', 'properties')
+    .where({ id: req.params.road_id })
+  .then(function([response]) {
+    if (response === undefined) {
+      return res(Boom.notFound());
+    }
+
+    return knex('road_properties')
+      .where({ id: req.params.road_id })
+      .update({ properties: applyPatch(response.properties, req.payload).newDocument });
```
Comment (mileswwatkins) [inline routes/road-properties.js]: Can't we do this without a library, using `Object.assign(response.properties, req.payload)` instead?

Or we could even rewrite both queries into just one, using the [JSONB concatenation operator](https://www.postgresql.org/docs/9.6/static/functions-json.html) (`||`), like:

```
knex.raw('
  UPDATE road_properties
  WHERE road_id = ${req.params.road_id}
  SET properties = properties || "${JSON.stringify(req.payload)}"::JSONB
')
```

Comment (jameslaneconkling) [reply routes/road-properties.js]: the PATCH spec isn't actually a matching JSON object of properties, but rather an array of operations: http://jsonpatch.com/

Comment (mileswwatkins) [reply routes/road-properties.js]: Gotcha, sounds good, then! Svelte to have one handler take care of all the CRUD.

File: routes/road-properties.js (hunk 1 above)
Comment (jameslaneconkling) [reply routes/road-properties.js]: added

**Pull request merged**